p2p/src/sync/peer.rs

```diff
@@ -291,13 +362,13 @@ where
             .is_none()
         {
             // TODO: Investigate this case. This can be used by malicious peers for a DoS attack.
-            self.request_headers(peer).await?;
+            self.request_headers().await?;
             return Ok(());
         }
 
         let header_ = header.clone();
         self.chainstate_handle.call(|c| c.preliminary_header_check(header_)).await??;
-        self.request_blocks(peer, vec![header])
+        self.request_blocks(vec![header])
     }
 
     /// Handles a result of message processing.
```

Comment:
This comment is probably outdated (fatal errors in the peer loop won't stop the whole p2p subsystem).